app-experimental.tex

```diff
@@ -2395,3 +2395,222 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
+\texttt{clc cs0, 0x100(ct0)}
+
+Second, a new data capability can be installed alongside the new PCC.
+This gives the compartment a capability ready to use in the register file.
+One option for this kind of compartment transition would be indirect sealed capability pairs.
+The following code example shows the brevity of this approach.
+The instruction sequence was shrunk to just one instruction reading from the new data capability(ct6).
+
+\texttt{clc cs0, 0(ct6)}
+
+\subsection{Sharing}
+
+Sealing explicitly forbids sharing capabilities between compartments.
+However, sometimes this behaviour is desired by software. We have designed two mechanisms for sharing capabilities between compartments:
+
+\begin{itemize}
+\item Explicit unsealing: This will explicitly unseal a capability and it can be shared either via the register file or via memory.
+Another compartment can pick up this capability and seal it again and use it.
+This does not protect against transitive capability leaks - an unsealed capability can be passed on to a third compartment and thus be leaked.
+\item CID spaces: CIDs can be separated into CID spaces (see Section~\ref{subsec:cid_spaces}).
+One approach is to allow unsealing within a CID space.
+This allows all compartments in that space to use that capability, but no other compartment can unseal the capability.
+This protects from transitive capability leaks outside of the CID space.
+As an addition, we envision a bit that -- if set -- allows sharing within the CID space.
+\end{itemize}
+
+Furthermore, we also envision the possibility of \textit{resealing} capabilities.
+A capability belonging to one compartment can be resealed to another compartment.
+This effectively means that code can seal one or more of its own capabilities for another compartment.
+The main advantages are that this avoids capabilities to be unsealed in the open as done with explicit unsealing as well as potential performance improvements avoiding the need to unseal and seal capabilites.
+However, resealing brings the disavantage for the receiving compartment of not knowing whether a capability was sealed by itself or another compartment.
+Therefore, we would expect the need for additional validation checks.
+
+\subsection{Explicit Unsealing}
+
+We need to reserve a CID value that represents an unsealed capability instead of a valid compartment ID.
+We can choose any value to represent an unsealed capability.
+In this proposal, a capability is unsealed if and only if  its ACID is set to 0.
+This ACID is also known as the zero ACID. A capability with the zero ACID can be CID sealed by any compartment.
+This introduces two new instructions (in comparison to the three-operand operations currently present in the CHERI-RISC-V, our proposed operations have two operands. The third implicit operand is PCC):
+
+\texttt{CCIDUnseal cd, cs1}\\
+If cs1.CID and PCC.CID match, then cs1 will be assigned to cd with cd.CID=0. Otherwise, throw an exception.
```

Comment:
Hopefully clear the tag rather than throwing an exception, to align with new tag-clearing philosophy? Is there a reason you need to trap here?

Reply:
Agreed, I think I tried to express: "this is not allowed" and chose to throw an exception. I has since escaped my proof-reads...